### DISCUSSION TITLE
RTSP with MJPEG codec - possibilities?

### DISCUSSION THREAD

**Discussion by ikcehcus** [bluenviron/mediamtx#601] [Questions]
I have a camera with a video codec set as MJPEG. I want to convert RTSP to HLS. I know that I can't do it directly using rtsp simple server and I must use ffmpeg. 
My solution for now is:

`"runOnDemand": "ffmpeg -rtsp_transport tcp -i 'rtsp://192.168.0.1:554/Streaming/Channels/102?transportmode=mcast&profile=Profile_2' -c:v libx264 -f rtsp rtsp://localhost:$RTSP_PORT/$RTSP_PATH",`

but this is an expensive solution. it takes 100 MB of RAM. 
Are there some other possibilities to convert RTSP to HLS when the video codec is MJPEG?

**Comment by aler9** [comment]
> Are there some other possibilities to convert RTSP to HLS when the video codec is MJPEG?

No, HLS is based on MPEG-TS, and MJPEG can't be packed into MPEG-TS:

https://stackoverflow.com/questions/30101380/packing-mjpeg-into-mpeg-ts

You need to re-encode.

[a comment by an automated account is not shown]
